public/js/test/sm-test-case/webpack.config.js

```diff
@@ -0,0 +1,8 @@
+
+module.exports = {
+  entry: "./entry.js",
+  output: {
+    filename: "bundle.js"
+  },
+  devtool: "sourcemap"
+}
```

Comment:
> I'd like the debugger test cases to all be hosted so that a user could go to them. Somewhat like mochitests examples as well.

That's the difference: these aren't integration tests like mochitests, they are unit tests. So it doesn't need runnable examples, it just needs some data. We have a bunch of really small data files like this in m-c too. I think it's worthwhile to have both because I can add any random file I need to without feeling like I need to tie it in somehow to the example's usage.

> I think there's a real benefit for contributors to be able to quickly load a test example and see how it works.

Yeah, I totally agree with more integration-y tests (as they need to be served anyway for those kinds of tests). Thinking about this more, maybe this is a better way of explaining it: what I need is just a bunch of source texts. Instead of saving fixtures, I'd like to have a folder with a bunch of sources (and some sourcemaps). It's just data that unit tests can use, and aren't grouped by various examples.

I will update this PR to reflect that, by renaming `sm-test-case` to something that is more just source texts. I could even make the thread client automatically load these as requested, so unit tests can freely load them.


> 
>     re: never changing
> 
> I don't see the examples changing too much, but if they do then I think it is okay to update the tests that are coupled.

Hopefully we'll have hundreds of unit tests so it really is a huge deal to change them. We've dealt with this on m-c and the unit tests data files are untouchable because you'd have to update almost every test. You can only add new test files, which is pretty easy anyway.

> 
> One thing to keep in mind is that it's important to be able to run the unit tests in the browser for devtools debugging. So, please try and make sure that the new tests can run with npm run mocha-server. I'm now wishing karma worked :/

Yeah, currently it won't work in the browser because it loads sourcemaps from disk. I don't find much value in karma to be honest because all this code doesn't even deal with the UI so it should run the same in all browsers, and I don't use it for debugging... To get these working though, we just need to fix up the utility functions to load stuff over the network instead of from disk. It won't be hard to get it running again when we want to focus on it. I'd rather not spend more time on it now though.